connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java

```diff
@@ -1512,11 +1773,25 @@ public WorkerTask doBuild(Task task,
         }
     }
 
-    // Visible for testing
+    /**
+     * Builds and returns an offset backing store for a regular source connector (i.e. when exactly-once support for source connectors is disabled).
+     * The offset backing store will either be just the worker's global offset backing store (if the connector doesn't define a connector-specific
+     * offset topic via its configs), just the connector-specific offset backing store (if the connector defines a connector-specific offsets
+     * topic which happens to be the same as the worker's global offset topic) or a combination of both the worker's global offset backing store
```

Comment:
Nit: the way we determine this is susceptible to edge cases where the topics aren't the same
```suggestion
     * topic which appears to be the same as the worker's global offset topic) or a combination of both the worker's global offset backing store
```